### ISSUE TITLE
iPython/pyreadline problem with 64-bit Python 3.5 on Windows 8.1 Pro

### ISSUE THREAD

**Issue by seth-p** [ipython/ipython#8815]
I just installed 64-bit Python 3.5.0 on Windows 8.1 Pro, and in installing all the packages I had working with 64-bit Python 3.4.3 I'm encountering the following problem in iPython (4.0.0) after installing pyreadline (2.0):
```
C:\Program Files\Python 3.5>ipython
Traceback (most recent call last):
  File "c:\program files\python 3.5\lib\runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\program files\python 3.5\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Python 3.5\Scripts\ipython.exe\__main__.py", line 9, in <module>
    sys.exit(start_ipython())
  File "c:\program files\python 3.5\lib\site-packages\IPython\__init__.py", line 118, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "c:\program files\python 3.5\lib\site-packages\traitlets\config\application.py", line 591, in launch_instance
    app.initialize(argv)
  File "<string>", line 2, in initialize
  File "c:\program files\python 3.5\lib\site-packages\traitlets\config\application.py", line 75, in catch_config_error
    return method(app, *args, **kwargs)
  File "c:\program files\python 3.5\lib\site-packages\IPython\terminal\ipapp.py", line 314, in initialize
    self.init_shell()
  File "c:\program files\python 3.5\lib\site-packages\IPython\terminal\ipapp.py", line 330, in init_shell
    ipython_dir=self.ipython_dir, user_ns=self.user_ns)
  File "c:\program files\python 3.5\lib\site-packages\traitlets\config\configurable.py", line 380, in instance
    inst = cls(*args, **kwargs)
  File "c:\program files\python 3.5\lib\site-packages\IPython\core\interactiveshell.py", line 531, in __init__
    self.init_readline()
  File "c:\program files\python 3.5\lib\site-packages\IPython\core\interactiveshell.py", line 1941, in init_readline
    import IPython.utils.rlineimpl as readline
  File "c:\program files\python 3.5\lib\site-packages\IPython\utils\rlineimpl.py", line 21, in <module>
    _rl = __import__(_rlmod_name)
  File "c:\program files\python 3.5\lib\site-packages\readline.py", line 6, in <module>
    from pyreadline.rlmain import Readline
  File "c:\program files\python 3.5\lib\site-packages\pyreadline\__init__.py", line 11, in <module>
    from . import unicode_helper, logger, clipboard, lineeditor, modes, console
  File "c:\program files\python 3.5\lib\site-packages\pyreadline\console\__init__.py", line 15, in <module>
    from .console import *
  File "c:\program files\python 3.5\lib\site-packages\pyreadline\console\console.py", line 610, in <module>
    msvcrt = cdll.LoadLibrary(ctypes.util.find_msvcrt())
  File "c:\program files\python 3.5\lib\ctypes\__init__.py", line 425, in LoadLibrary
    return self._dlltype(name)
  File "c:\program files\python 3.5\lib\ctypes\__init__.py", line 347, in __init__
    self._handle = _dlopen(self._name, mode)
TypeError: bad argument type for built-in operation

If you suspect this is an IPython bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@scipy.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    c.Application.verbose_crash=True
```

Without pyreadline, iPython runs (though doesn't support tab completion...):
```
C:\Program Files\Python 3.5>ipython
WARNING: Readline services not available or not loaded.
WARNING: Proper color support under MS Windows requires the pyreadline library.
You can find it at:
http://ipython.org/pyreadline.html

Defaulting color scheme to 'NoColor'
Python 3.5.0 (v3.5.0:374f501f4567, Sep 13 2015, 02:27:37) [MSC v.1900 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 4.0.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]:
```

Any idea what I might be doing wrong, or is this a bug in pyreadline (or iPython)? If pyreadline doesn't work with Python 3.5, I guess I'll revert to 3.4.

For what it's worth, these are all the packages I have installed:
```
C:\Program Files\Python 3.5>pip3 list
alchy (2.0.1)
cvxopt (1.1.7)
cvxpy (0.2.28)
Cython (0.23.2)
dateutils (0.6.6)
decorator (4.0.2)
ecos (2.0.4)
ftputil (3.2)
ipykernel (4.0.3)
ipython (4.0.0)
ipython-genutils (0.1.0)
Jinja2 (2.8)
jsonschema (2.5.1)
jupyter-client (4.0.0)
jupyter-core (4.0.4)
MarkupSafe (0.23)
matplotlib (1.4.3)
mistune (0.7.1)
Mosek (7.1.33)
nbconvert (4.0.0)
nbformat (4.0.0)
notebook (4.0.4)
numexpr (2.4.4)
numpy (1.9.2)
pandas (0.17.0rc1)
path.py (8.1.1)
patsy (0.4.0)
pickleshare (0.5)
PICOS (1.1.1)
pip (7.1.2)
psutil (3.2.1)
pydash (3.3.0)
Pygments (2.0.2)
pyparsing (2.0.3)
pyreadline (2.0)
python-dateutil (2.4.2)
pytz (2015.4)
pyzmq (14.7.0)
scipy (0.16.0)
scs (1.1.4)
setuptools (18.3.1)
simplegeneric (0.8.1)
six (1.9.0)
sortedcontainers (0.9.6)
SQLAlchemy (1.0.8)
statsmodels (0.6.1)
tables (3.2.1)
toolz (0.7.4)
tornado (4.2.1)
traitlets (4.0.0)
```

**Comment by Carreau** [comment]
I think that pyreadline + python 3.5 is known not to be working well together. 
Could you try a conda + 3.5 python env? I think it might have more chances of working.

**Comment by seth-p** [comment]
I'm afraid I'm not going to be able to try conda any time soon.

**Comment by seth-p** [comment]
This is fixed with pyreadline-2.1.

**Comment by Carreau** [comment]
Thanks for the info !
